public/app/core/reducers/navModel.ts

```diff
@@ -60,6 +62,19 @@ export const navIndexReducer = (state: NavIndex = initialState, action: AnyActio
     }
 
     return { ...state, ...newPages };
+  } else if (updateConfigurationSubtitle.match(action)) {
+    const newCfgState = { ...state.cfg, subTitle: `Organization: ${action.payload}` };
+    // Update configuration (cfg) and all configuration children navItems
+    return {
+      ...state,
+      cfg: newCfgState,
+      datasources: { ...state.datasources, parentItem: newCfgState },
+      users: { ...state.users, parentItem: newCfgState },
+      teams: { ...state.teams, parentItem: newCfgState },
+      plugins: { ...state.plugins, parentItem: newCfgState },
+      'org-settings': { ...state['org-settings'], parentItem: newCfgState },
+      apikeys: { ...state.apikeys, parentItem: newCfgState },
+    };
```

Comment:
It looks like we're replacing the complete parentItem every line here. I think something like this is the safer way to mutate this state, wdyt @KrammerJake 
```suggestion
         const subTitle = `Organization: ${action.payload}`;
     
         return {
      ...state,
      cfg: { ...state.cfg, subTitle },
      datasources: getItemWithNewSubTitle(state.datasources, subTitle),
      users: getItemWithNewSubTitle(state.users, subTitle),
      teams: getItemWithNewSubTitle(state.teams, subTitle),
      plugins: getItemWithNewSubTitle(state.plugins, subTitle),
      'org-settings': getItemWithNewSubTitle(state['org-settings'], subTitle),
      apikeys: getItemWithNewSubTitle(state.apikeys, subTitle),
    };

    };
    
    // add this outside the reducer
    const getItemWithNewSubTitle = (item: NavModelItem, subTitle: string): NavModelItem => {
  return { ...item, parentItem: { ...item.parentItem, text: item.parentItem?.text ?? '', subTitle } };
};
```

Reply:
I agree, your way seems safer and more precise.